plasma/plasma-mvp/morevp-alice-bob.md

```diff
@@ -0,0 +1,103 @@
+MoreVP Alice-Bob Scenarios
+---
+
+Note that these scenarios work with N participants, not just limited to 1 input 1 output. 
+
+Also, note that in all of these scenarios the exits are given the priority of the youngest input to the transaction.
+
+#### Alice & Bob are honest and cooperating:
+
+Happy path.
+
+1. Alice spends UTXO1 in TX1 to Bob.
+2. Operator withholds, neither Alice nor Bob know if the transaction has been included. 
+3. Someone with access to TX1 (Alice, Bob, or otherwise) starts an exit and places `exit bond`.
+4. Alice and Bob "piggyback" onto the exit and each put up `piggyback bond`.
```

Comment:
let's explicitly say what if they don't or just one of them does? This touches upon the subject of resolving whether the transaction happened, which I still cannot wrap my head around. So maybe let's try list these sub-scenarios explicitly.

Reply:
ok gotcha, but let's get super-specific.
A) Alice and Bob piggyback - covered
B) only Alice piggybacks - ?
C) only Bob piggybacks -?
D) Alice1/2 and Bob1/2 piggyback -?
E) Alice1, Bob1/2 piggyback, Alice2 doesnt -?
...

Reply:
I don't think it's particularly useful to go through each of these cases. Whether or not a user piggybacks has no impact on whether the transaction is valid (outputs get to exit) or invalid (inputs get to exit). That decision flow is determined by the following logic:

![decision flow](https://camo.githubusercontent.com/66b925344ed269f2907adc8685403750626b1c01/68747470733a2f2f692e696d6775722e636f6d2f696f5965674f422e706e67)

If the owner of an input piggybacks and it's discovered that the transaction is valid, then that owner will be refunded. Similarly, if the owner of an output piggybacks and it's discovered that the transaction is invalid, then the owner of that output will be refunded. It's just a way to ensure that every input/output that might be exitable has an associated bond to slash.

Reply:
All 3 comments on that topic were extremely clarifying. Thank you! Now I agree no need to list the scenarios nor state machine, but:

> Because it's not known in advance whether the transaction is valid or not, owners of both inputs and outputs have to piggyback preemptively.

and

> Whether or not a user piggybacks has no impact on whether the transaction is valid (outputs get to exit) or invalid (inputs get to exit).

I think that these two statements should be included explicitly in the description of the exit game (maybe others as well). Also please review the definition and usage of "valid"/"invalid" transaction, thourghout the paper and here. I think sometimes the usage isn't too strict, hence increased chance of confusion. Maybe a different term then - something that implies directly, that the thing is about whether inputs or outputs (but never a mix) exit - `done`/`cancelled`? `completed/reverted`? 

Also a minor nitpick: note that your previous comment was a different thing :) : 

> Piggybacking states the following things: (1) "This transaction is valid"